libs/dal/src/repositories/topic/topic.repository.ts

```diff
@@ -0,0 +1,30 @@
+import { AuthProviderEnum } from '@novu/shared';
+import { FilterQuery } from 'mongoose';
+
+import { EnvironmentId, OrganizationId, TopicKey, TopicEntity } from './topic.entity';
+import { Topic } from './topic.schema';
+
+import { BaseRepository } from '../base-repository';
+
+type IPartialTopicEntity = Omit<TopicEntity, '_environmentId' | '_organizationId'>;
+
+type EnforceEnvironmentQuery = FilterQuery<IPartialTopicEntity> &
+  ({ _environmentId: EnvironmentId } | { _organizationId: OrganizationId });
+
+export class TopicRepository extends BaseRepository<EnforceEnvironmentQuery, TopicEntity> {
+  constructor() {
+    super(Topic, TopicEntity);
+  }
+
+  async findTopicByKey(
+    key: TopicKey,
+    organizationId: OrganizationId,
+    environmentId: EnvironmentId
```

Comment:
What do you think about creating a container for these parameters? With all the new DAL enforcement I started to have doubts about primitive params on methods, especially strings, of course, your decision and if you like the idea you could wrap it on the next PR does not have to be in this one. :)

Reply:
In following PRs I moved them to a common types file for the topics repository and I would like in the future start to replace in API the string primitives for this kind of type wrappers. I think we should discuss it in the team where would be the best place to host them and how to export them to be able to be reused. But definitely we are on the same page. 🙂